types/lib/nodecg-instance.d.ts

```diff
@@ -0,0 +1,106 @@
+import {IRouter, RequestHandler} from 'express-serve-static-core';
+
+import {Logger} from './logger';
+import {ReplicantOptions, Replicant} from './replicant';
+import {NodeCGConfig} from './config';
+import {SendMessageReturnType} from './nodecg-static';
+import {Platform} from './platform';
+
+interface NodeCGCommon<P extends Platform, M = SendMessageReturnType<P>> {
+	bundleName: string;
+	bundleConfig: any;
+	bundleVersion: string;
+	readonly bundleGit: {
+		branch: string;
+		hash: string;
+		shortHash: string;
+		date?: Date;
+		message?: string;
+	};
+	Logger: typeof Logger;
+	log: Logger;
+	readonly config: NodeCGConfig;
+	sendMessage(
+		messageName: string,
+		cb?: (error: any, ...args: any[]) => void
+	): M;
+	sendMessage(
+		messageName: string,
+		data: any,
+		cb?: (error: any, ...args: any[]) => void
+	): M;
+	listenFor(messageName: string, handlerFunc: (message: any) => void): void;
+	listenFor(
+		messageName: string,
+		bundleName: string,
+		handlerFunc: (message: any) => void
```

Comment:
These `listenFor` typedefs do not include the `acknowledgement` callback. It is provided as the second arg to the callback (server-side only).

Reply:
Added with `handled` support